src/server/bg_workers.js

```diff
@@ -84,12 +84,13 @@ function run_master_workers() {
         run_immediate: true
     }, stats_aggregator.background_worker);
 
-    register_bg_worker(new NamespaceMonitor({
-        name: 'namespace_monitor',
-        client: server_rpc.client,
-        should_monitor: nsr => !nsr.nsfs_config,
-    }));
-
+    if (!config.DISABLE_NAMESPACE_MONITOR) {
+        register_bg_worker(new NamespaceMonitor({
+            name: 'namespace_monitor',
+            client: server_rpc.client,
+            should_monitor: nsr => !nsr.nsfs_config,
+        }));
+    }
```

Comment:
I would suggest to change the variable name to align with the code style of this file: From `config.DISABLE_NAMESPACE_MONITOR` to `config.NAMESPACE_MONITOR_ENABLED`
and then:
```suggestion
    if (config.NAMESPACE_MONITOR_ENABLED) {
        register_bg_worker(new NamespaceMonitor({
            name: 'namespace_monitor',
            client: server_rpc.client,
            should_monitor: nsr => !nsr.nsfs_config,
        }));
    } else {
        dbg.warn('NAMESPACE_MONITOR NOT ENABLED');
    }
```

Reply:
thanks, updated